zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/ManagedInterpreterGroup.java

```diff
@@ -141,17 +165,27 @@ private void close(Collection<Interpreter> interpreters) {
   private void closeInterpreter(Interpreter interpreter) {
     Scheduler scheduler = interpreter.getScheduler();
 
-    for (final Job job : scheduler.getAllJobs()) {
-      job.abort();
-      job.setStatus(Job.Status.ABORT);
-      LOGGER.info("Job " + job.getJobName() + " aborted ");
-    }
+    // Need to abort the task being executed
+    // when actively shutting down the remote interpreter
+    if (false == remoteIntpProcessIsShutdown()) {
```

Comment:
this isn't quite I'd expected... if the process is created but it's not running, yes it could be being shutdown else where but it could also be that it was killed or crashed or something. in the latter case, shouldn't it go ahead to abort the jobs anyway?

Reply:
### 1. If interpreter process is normal, isRunning() must equal true
`isRunning()` is not a flag, It's a function call.
`interpreterProcess.isRunning()` Will call `RemoteInterpreterUtils.java::checkIfRemoteEndpointAccessible()` , Check if the remote interpreter is available through the socket.

### 2. Avoid invalid calls
`Job.abort()` Will go to the remote to call the interpreter, `Interpreter.cancel(getInterpreterContext(null));` function. If the interpreter is no longer connected, The `interpreter.cancel()` function cannot be called correctly.

But if you find that the remote interpreter is not available in `job.abort()`, Will pass again
`RemoteInterpreter.java::getOrCreateInterpreterProcess()`, Trying to create an interpreter process, This will fail due to a 30 second timeout.